src/public/js/main.js

```diff
@@ -15,16 +17,33 @@
       }
 
       var msg = $('#post').val();
-      var encoded = RS.encode(msg);
+      var i = 1;
+      var last = 0;
+      var encoded = new Array(0);
+      for (; i <= msg.length; i++) {
+        if (i % k == 0) {
+          Array.prototype.push.apply(encoded, RS.encode(msg.slice(i - k, i)));
```

Comment:
Basically the annoying thing here is that the output of encode is length n and the input is length k. so the indexes get really out of whack if you try looping setting values of a pre-allocated array. Im sure it is doable, but It is easy to just append.